src/import/prepare.rs

```diff
@@ -109,22 +108,51 @@ impl<Client: ImportClient> PrepareJob<Client> {
             }
 
             let range = RangeInfo::new(&start, k, ctx.raw_size());
-            if let Ok(true) = self.run_prepare_job(range) {
+            if let Ok(true) = self.run_prepare_job(range, &mut wait_scatter_regions) {
                 num_prepares += 1;
             }
 
             start = k.to_owned();
             ctx.reset(k);
         }
 
-        num_prepares
+        // We need to wait all regions for scattering finished.
+        let start = Instant::now();
+        while let Some(region_id) = wait_scatter_regions.pop() {
+            let start = Instant::now();
+            for i in 0..SCATTER_WAIT_MAX_RETRY_TIMES {
+                if self.client.is_scatter_region_finished(region_id)? {
```

Comment:
meet error also can try to scatter next region.

Reply:
It's OK, just ensure most of the regions have been scattered.

Reply:
but now it will return directly if meet error?

Reply:
You could reference the definition of `is_scatter_region_finished`, we should interrupt the current import task if some errors that we could not handle occurred.

Reply:
got it.